### PR TITLE
update cdi config docs

### PR DESCRIPTION
**Pull request by mhenriks**
Signed-off-by: Michael Henriksen <mhenriks@redhat.com>

**What this PR does / why we need it**:

Update docs to reference setting on `CDI`.  Also added examples.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/core/v1beta1/types.go [hunk 1]
```diff
@@ -377,7 +377,7 @@ type CDIConfigSpec struct {
 	// FilesystemOverhead describes the space reserved for overhead when using Filesystem volumes. A value is between 0 and 1, if not defined it is 0.055 (5.5% overhead)
 	FilesystemOverhead *FilesystemOverhead `json:"filesystemOverhead,omitempty"`
 	// Preallocation controls whether storage for DataVolumes should be allocated in advance.
-	Preallocation bool `json:"preallocation,omitempty"`
+	Preallocation *bool `json:"preallocation,omitempty"`
```
Comment (maya-r) [inline pkg/apis/core/v1beta1/types.go:380]: Is this change intentional? it seems unrelated.

Comment (mhenriks) [reply pkg/apis/core/v1beta1/types.go:380]: Intentional and somewhat related as I was reviewing the config params.  This property is currently unreferenced, though.  But I assume @tomob has plans for it.

Comment (tomob) [reply pkg/apis/core/v1beta1/types.go:380]: What do you mean "unreferenced"? What about that? https://github.com/kubevirt/containerized-data-importer/blob/d31591de94dc2c210b55cf6755fb55fb11397e64/pkg/controller/util.go#L639

Comment (mhenriks) [reply pkg/apis/core/v1beta1/types.go:380]: @tomob I updated the spec, not the status

Comment (tomob) [reply pkg/apis/core/v1beta1/types.go:380]: Oh, crap.

What I actually wanted to use there is the spec. It cased the bug I talked about today: https://github.com/kubevirt/containerized-data-importer/pull/1565

Comment (mhenriks) [reply pkg/apis/core/v1beta1/types.go:380]: @tomob so do you want to change to `*bool` in your PR? Or keep here?

Comment (mhenriks) [reply pkg/apis/core/v1beta1/types.go:380]: It should be `*bool` because it is optional.  The status field can remain `bool`

Comment (awels) [reply pkg/apis/core/v1beta1/types.go:380]: Also make sure to run make generate, this changes the generated deep copy

**Comment by awels** [comment]
I think we should fix the *bool in #1565 especially since that needs to get fixed in some other way.

**Comment by mhenriks** [comment]
@awels I reverted the change to `preallocation`
